syft/workers/message_handler.py

```diff
@@ -196,16 +196,19 @@ def respond_to_obj_req(self, msg: ObjectRequestMessage):
         obj_id = msg.object_id
         user = msg.user
         reason = msg.reason
+        get_copy = msg.get_copy
 
         obj = self.get_obj(obj_id)
 
         permitted = all(map_chain_call(obj, "allow", user=user))
         if not permitted:
             raise GetNotPermittedError()
-        else:
+        elif not get_copy:
             self.object_store.de_register_obj(obj)
             return obj
```

Comment:
NITPICK: Remove this since you return the object at the end

Reply:
oops, done